jdaviz/utils.py

```diff
@@ -389,8 +389,11 @@ def flux_conversion(values, original_units, target_units, spec=None, eqv=None, s
         # the unit of the data collection item object, could be flux or surface brightness
         spec_unit = str(spec.flux.unit)
 
-        # Need this for setting the y-limits
-        eqv = u.spectral_density(spectral_values)
+        # Need this for setting the y-limits but values from viewer might be downscaled
+        if len(values) == spectral_values.size:
+            eqv = u.spectral_density(spectral_values)
+        else:
+            eqv = u.spectral_density(spec.spectral_axis[0])
```

Comment:
I actually not sure if grabbing `[0]` is entirely correct but I also don't know what is really happening when `values` are only an array of 7 elements. I think that is the downsampled version from the viewer but really hard to tell for sure.

Reply:
I added value checking to Clare's test, so hopefully good enough but still not sure what situation would enter this block. But if I don't add this block, you get the shape mismatch error. 🤷‍♀️